wpiunits/generate_units.py

```diff
@@ -395,10 +385,32 @@ def main():
             helpers=helpers,
         )
 
-        output(f"{rootPath}/measure", f"{unit_name}.java", interfaceContents)
-        output(f"{rootPath}/measure", f"Immutable{unit_name}.java", immutableContents)
-        output(f"{rootPath}/measure", f"Mut{unit_name}.java", mutableContents)
+        output(rootPath / "measure", f"{unit_name}.java", interfaceContents)
+        output(rootPath / "measure", f"Immutable{unit_name}.java", immutableContents)
+        output(rootPath / "measure", f"Mut{unit_name}.java", mutableContents)
+
+
+def main(argv):
+    script_path = Path(__file__).resolve()
+    dirname = script_path.parent
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--output_directory",
+        help="Optional. If set, will output the generated files to this directory, otherwise it will use a path relative to the script",
+        default=dirname / "src/generated",
+        type=Path,
+    )
+    parser.add_argument(
+        "--template_root",
+        help="Optional. If set, will use this directory as the root for the jinja templates",
+        default=dirname / "src/generate",
+        type=Path,
+    )
+    args = parser.parse_args(argv)
+
+    generate_units(args.output_directory, args.template_root)
 
 
 if __name__ == "__main__":
-    main()
+    main(sys.argv[1:])
```

Comment:
Passing argv here isn't necessary. wpiformat just calls `parser.parse_args()`, which automatically looks at `sys.argv`.

Reply:
All the other generation scripts and upstream_utils do this

Reply:
They shouldn't.